website/project/model.py

```diff
@@ -1313,6 +1360,13 @@ def nodes_primary(self):
             if node.primary
         ]
 
+    def node_and_primary_descendants(self):
+        """Gets an iterator for a node and all of its primary (non-pointer) descendants
```

Comment:
Minor, but we follow [PEP 257](https://www.python.org/dev/peps/pep-0257/) for docstring conventions.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".